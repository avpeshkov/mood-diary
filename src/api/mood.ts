import { MoodObject } from "types/mood";
import { db } from "services/firebase";
import { getCurrentUserUid } from "helpers/auth";
import firebase from "firebase";

const MOOD_LIST_PATH = "user-mood-list";

function getMoodList(): Promise<firebase.database.DataSnapshot> {
    const userUid = getCurrentUserUid();
    return db.ref(`${MOOD_LIST_PATH}/${userUid}`).get();
}

const postPatchMood = (moodObject: MoodObject): Promise<unknown> => {
    const userUid = getCurrentUserUid();
    const updatedObject = { ...moodObject, date: moodObject.date.toString() };
    if (moodObject.id) {
        return db.ref(`${MOOD_LIST_PATH}/${userUid}/${moodObject.id}`).update(updatedObject);
    } else {
        return (db.ref(`${MOOD_LIST_PATH}/${userUid}/`).push(updatedObject) as unknown) as Promise<unknown>;
    }
};

const deleteMood = (moodId: number): Promise<unknown> => {
    const userUid = getCurrentUserUid();
    return db.ref(`user-mood-list/${userUid}/${moodId}`).remove();
};

const MoodApi = {
    getMoodList,
    postPatchMood,
    deleteMood,
};

export default MoodApi;
