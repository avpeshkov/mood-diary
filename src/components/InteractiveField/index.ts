export * from "./InteractiveField";
export * from "./components";
