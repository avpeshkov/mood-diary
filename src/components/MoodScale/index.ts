export * from "./MoodScale";
export * from "./components";
