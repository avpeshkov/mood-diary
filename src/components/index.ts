export * from "./FooterComponent";
export * from "./HeaderComponent";
export * from "./ScreenWrapper";
export * from "./AuthorizedCheckHoc";
