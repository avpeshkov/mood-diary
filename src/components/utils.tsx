export const getRandomIndex = (arrayLength: number): number => {
    return Math.floor(Math.random() * Math.floor(arrayLength));
};
