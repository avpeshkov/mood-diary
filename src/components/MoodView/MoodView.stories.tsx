import React from "react";
import { withKnobs, date, number } from "@storybook/addon-knobs";
import { Mood } from "../MoodScale/components/MoodItem";
import { MoodView } from "./MoodView";

export default {
    title: "MoodView",
    decorators: [withKnobs],
};

export const simpleData = () => (
    <MoodView
        moodObject={{
            date: new Date(date("date", new Date("December 16, 2020 03:24:00"))),
            mood: number("mood", 7) as Mood,
            comment: "Comment about my life",
        }}
    />
);

export const withoutComment = () => (
    <MoodView
        moodObject={{
            date: new Date(date("date", new Date("December 16, 2020 03:24:00"))),
            mood: number("mood", 7) as Mood,
            comment: "",
        }}
    />
);
