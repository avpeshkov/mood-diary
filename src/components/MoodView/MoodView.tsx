import React from "react";
import styled from "@emotion/styled";
import { MoodScale } from "../MoodScale";
import { MoodObject } from "types/mood";
import "react-datepicker/dist/react-datepicker.css";
import { css } from "@emotion/core";

interface MoodViewProps {
    moodObject: MoodObject;
    updateMoodObject?: (moodId: number, action: "edit" | "delete") => void;
}

const MoodViewWrapper = styled.div`
    display: flex;
    flex-direction: row;
    padding-bottom: 10px;
    padding-top: 10px;
    width: 500px;
    border-bottom: 2px solid lightgray;
    justify-content: space-between;
`;

const MoodContentWrapper = styled.div`
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    margin-right: 10px;
`;

const MoodEditButtonsWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
`;

const ViewWrapper = styled.div`
    display: inline-flex;
    flex-direction: row;
    margin-bottom: 10px;
`;

export const DateWrapper = styled.div`
    display: inline-flex;
    margin-bottom: 5px;
    font-weight: bold;
`;
const ActionButton = styled.button`
    display: inline-flex;
    background-color: darkgreen;
    border-radius: 5px;
    border: 0;
    color: white;
    width: fit-content;
    align-items: center;
    align-self: flex-end;
    text-align: center;
    ${(props: { backgroundColor?: string }) =>
        props.backgroundColor &&
        css`
            background-color: ${props.backgroundColor};
        `}
`;

// Комнпонент для обображения/редактирования основной модели прилоежния.

export const MoodView: React.FC<MoodViewProps> = (props: MoodViewProps) => {
    const { moodObject, updateMoodObject } = props;

    return (
        <MoodViewWrapper>
            <MoodContentWrapper>
                <ViewWrapper>
                    <label>Date: </label>
                    <DateWrapper> {moodObject.date.toDateString()}</DateWrapper>
                </ViewWrapper>
                <MoodScale currentMood={moodObject.mood} />
                {moodObject.comment && (
                    <>
                        <label>Comment:</label>
                        <span id="comment">{moodObject.comment}</span>
                    </>
                )}
            </MoodContentWrapper>
            {moodObject.id && updateMoodObject && (
                <MoodEditButtonsWrapper>
                    <ActionButton onClick={() => updateMoodObject(moodObject.id!, "edit")}>Edit</ActionButton>
                    <ActionButton onClick={() => updateMoodObject(moodObject.id!, "delete")} backgroundColor="red">
                        Delete
                    </ActionButton>
                </MoodEditButtonsWrapper>
            )}
        </MoodViewWrapper>
    );
};
