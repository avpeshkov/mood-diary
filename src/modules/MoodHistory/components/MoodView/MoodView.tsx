import React from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/core";
import { MoodScale } from "modules/MoodHistory/components/MoodScale";
import { MoodObject } from "modules/MoodHistory/types";

const MoodViewWrapper = styled.div`
    display: flex;
    flex-direction: row;
    padding-bottom: 10px;
    padding-top: 10px;
    width: 500px;
    border-bottom: 2px solid lightgray;
    justify-content: space-between;
`;

const MoodContentWrapper = styled.div`
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    margin-right: 10px;
`;

const MoodEditButtonsWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
`;

const ViewWrapper = styled.div`
    display: inline-flex;
    flex-direction: row;
    margin-bottom: 10px;
`;

export const DateWrapper = styled.div`
    display: inline-flex;
    margin-bottom: 5px;
    font-weight: bold;
    padding-left: 10px;
`;
const ActionButton = styled.button`
    display: inline-flex;
    background-color: #6180da;
    border-radius: 5px;
    border: 0;
    color: white;
    width: fit-content;
    align-items: center;
    align-self: flex-end;
    text-align: center;
    ${(props: { backgroundColor?: string }) =>
        props.backgroundColor &&
        css`
            background-color: ${props.backgroundColor};
        `}
`;
const Comment = styled.span`
    margin-top: 10px;
    box-sizing: border-box;
    white-space: nowrap;
    overflow-wrap: break-word;
    word-break: break-word;
    line-height: 18px;
    overflow: hidden;
    text-overflow: ellipsis;
    width: 400px;
`;

interface MoodViewProps {
    moodObject: MoodObject;
    editMoodObject?: (moodId: string) => void;
    deleteMoodObject?: (moodId: string) => void;
    style?: React.CSSProperties;
}

/** Комнпонент для обображения/редактирования основной модели прилоежния. */
export const MoodView: React.FC<MoodViewProps> = (props: MoodViewProps) => {
    const { moodObject, editMoodObject, deleteMoodObject, style } = props;

    return (
        <MoodViewWrapper style={style} id="mood-view-wrapper">
            <MoodContentWrapper>
                <ViewWrapper>
                    <label>Date: </label>
                    <DateWrapper> {moodObject.date.split("GMT")[0]}</DateWrapper>
                </ViewWrapper>
                <MoodScale currentMood={moodObject.mood} />
                {moodObject.comment && (
                    <Comment id="comment" title={moodObject.comment}>
                        <label>Comment: </label>
                        {moodObject.comment}
                    </Comment>
                )}
            </MoodContentWrapper>
            {moodObject.id && (
                <MoodEditButtonsWrapper>
                    {editMoodObject && (
                        <ActionButton data-test-id="edit-mood-btn" onClick={() => editMoodObject(moodObject.id!)}>
                            Edit
                        </ActionButton>
                    )}
                    {deleteMoodObject && (
                        <ActionButton data-test-id="delete-mood-btn" onClick={() => deleteMoodObject(moodObject.id!)} backgroundColor="#ff6464">
                            Delete
                        </ActionButton>
                    )}
                </MoodEditButtonsWrapper>
            )}
        </MoodViewWrapper>
    );
};
