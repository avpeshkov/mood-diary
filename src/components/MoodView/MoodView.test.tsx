import React from "react";
import snapshotDiff from "snapshot-diff";

import { shallow } from "enzyme";
import { MoodView } from "components/MoodView";
import { Mood } from "types/mood";

describe("MoodView", () => {
    const moodObject = { mood: 5 as Mood, comment: "test", date: new Date("December 16, 2020 03:24:00") };
    test("no snapshot difference between equal components", () => {
        expect(snapshotDiff(<MoodView moodObject={moodObject} />, <MoodView moodObject={moodObject} />)).toMatchSnapshot();
    });
    const moodObjectWithoutComment = { ...moodObject, comment: "" };
    test("no snapshot difference between components", () => {
        expect(snapshotDiff(<MoodView moodObject={moodObjectWithoutComment} />, <MoodView moodObject={moodObjectWithoutComment} />)).toMatchSnapshot();
    });

    it("renders component with comment", () => {
        const view = shallow(<MoodView moodObject={moodObject} />);

        expect(view.find("#comment").length).toEqual(1);
    });
    it("renders component without comment", () => {
        const scale = shallow(<MoodView moodObject={moodObjectWithoutComment} />);
        expect(scale.find("#comment").length).toEqual(0);
    });
});
