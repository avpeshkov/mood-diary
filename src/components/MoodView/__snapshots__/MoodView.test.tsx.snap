// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodView no snapshot difference between components 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`MoodView no snapshot difference between equal components 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
